inference-engine/src/auto_plugin/auto_async_infer_request.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+#include <string>
+#include <map>
+
+#include "auto_async_infer_request.hpp"
+
+namespace AutoPlugin {
+    using namespace InferenceEngine;
+
+AutoAsyncInferRequest::AutoAsyncInferRequest(
+    const AutoInferRequest::Ptr&                inferRequest,
+    const bool                                  needPerfCounters,
+    const AutoExecutableNetwork::Ptr&           autoExecutableNetwork,
+    const ITaskExecutor::Ptr&                   callbackExecutor) :
+    AsyncInferRequestThreadSafeDefault(inferRequest, nullptr, callbackExecutor),
+    _autoExecutableNetwork{autoExecutableNetwork},
+    _inferRequest{inferRequest},
+    _needPerfCounters{needPerfCounters} {
+    _pipeline = {
+        {_autoExecutableNetwork, [this]{_inferRequest->SetBlobsToRealInferRequest();}},
+        {_autoExecutableNetwork, [this]{_inferRequest->Infer();}}
+    };
+}
+
+void AutoAsyncInferRequest::Infer_ThreadUnsafe() {
+    InferUsingAsync();
```

Comment:
like the default impl already suffice

Reply:
Yes, or no, the default Infer_ThreadUnsafe() will run _syncPipeline but not _pipeline. the InferUsingAsync() will run _pipeline. 
I have modified to use default impl by changing _pipeline to _syncPipeline, please correct me if anything unexpected such as performance or something..

Reply:
I just find a bug when running benchmark_app, since it will call StartAsync() -> StartAsync_ThreadUnsafe() -> RunFirstStage for _pipeline. I guess we have to override Infer_ThreadUnsafe() so that both Infer() and StartAsync() could use _pipeline. Or override StartAsync_ThreadUnsafe() to use _syncPipeline.